From 4823fb38573d55dc918c2c891a81362bcfaa09e3 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Tue, 26 Nov 2013 13:49:00 +0800
Subject: [PATCH 478/502] ENGR00215174 update to pass build for android 4.4

property_get() function limit the input buffer size to more than 92 bytes.
So change sensor and orient property buffsize from 32 to 92 bytes.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera/CameraModule.cpp | 2 +-
 mx6/libcamera/CameraUtil.h     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera/CameraModule.cpp b/mx6/libcamera/CameraModule.cpp
index 11081b6..f32ce53 100644
--- a/mx6/libcamera/CameraModule.cpp
+++ b/mx6/libcamera/CameraModule.cpp
@@ -677,7 +677,7 @@ static void GetCameraPropery(char *pFaceBackCameraName,
                              int  *pFaceBackOrient,
                              int  *pFaceFrontOrient)
 {
-    char orientStr[10];
+    char orientStr[92];
 
     property_get(FACE_BACK_CAMERA_NAME,
                  pFaceBackCameraName,
diff --git a/mx6/libcamera/CameraUtil.h b/mx6/libcamera/CameraUtil.h
index 91de557..a9924fa 100644
--- a/mx6/libcamera/CameraUtil.h
+++ b/mx6/libcamera/CameraUtil.h
@@ -76,7 +76,7 @@ using namespace android;
 #define MAX_CAPTURE_BUFFER      3
 #define DISPLAY_WAIT_TIMEOUT    5000
 #define CAMAERA_FILENAME_LENGTH 256
-#define CAMERA_SENSOR_LENGTH    32
+#define CAMERA_SENSOR_LENGTH    92
 #define CAMERA_FORMAT_LENGTH    32
 #define CAMER_PARAM_BUFFER_SIZE 512
 #define PARAMS_DELIMITER ","
-- 
1.8.0

